sdk/core/src/policies/retry_policies/exponential_retry.rs

```diff
@@ -0,0 +1,86 @@
+use crate::policies::{Context, Policy, PolicyResult, Request, Response};
+use crate::sleep::sleep;
+use chrono::{DateTime, Local};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// Retry policy with exponential backoff.
+///
+/// Retry policy with exponential backoff (with an added random delay up to 256 ms). Each retry
+/// will happen at least after an exponential wait time. So if x is the first retry wait, the
+/// second will be x*2, the third x*4 and so on. The policy will retry until the maximum number of
+/// retries have been reached or the maximum allowed delay has passed (whichever comes first). The
+/// wait time is not precise.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct ExponentialRetryPolicy {
+    delay: Duration,
+    max_retries: u32,
+    max_delay: Duration,
+}
+
+impl Default for ExponentialRetryPolicy {
+    fn default() -> Self {
+        Self {
+            delay: Duration::from_secs(3),
+            max_retries: 3,
+            max_delay: Duration::from_secs(30),
+        }
+    }
+}
+
+impl ExponentialRetryPolicy {
+    fn is_expired(
+        &self,
+        first_retry_time: &mut Box<Option<DateTime<Local>>>,
+        current_retries: &u32,
```

Comment:
mmm no, the idea of the code is to update the passed `first_retry_time`. This way the following function call knows that this is not the first retry. I could have returned a tuple `(bool, DateTime<Local>)` but I think it would have been harder to understand. 

I could have used `*first_retry_time = Box::new(Some(Local::now()))` instead of `mem::swap`. Maybe that way is more expressive?

Reply:
I'm referring to `current_retries` not `first_retry_time`